### PR TITLE
ci: update the workflow

### PR DESCRIPTION
**Pull request by joshuachp**
Add update versions for the workflows, use reusable workflows to have a single entry point. Check if the DCO hash passed before running CI.

### CODE REVIEW

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          components: clippy
+      - name: Install sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+      - name: Install Protoc
+        run: |
+          curl -LO $PB_REL/download/v22.2/protoc-22.2-linux-x86_64.zip
+          unzip protoc-22.2-linux-x86_64.zip -d $HOME/.local
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
```
Comment (harlem88) [inline .github/workflows/check.yaml:62]: Is it still useful? protoc should be used only on message-hub-proto repo.

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        uses: dtolnay/rust-toolchain@stable
+      - name: Install sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+      - name: cargo install cargo-hack
+        uses: taiki-e/install-action@cargo-hack
+      - name: Install Protoc
+        run: |
+          curl -LO $PB_REL/download/v22.2/protoc-22.2-linux-x86_64.zip
+          unzip protoc-22.2-linux-x86_64.zip -d $HOME/.local
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
```
Comment (harlem88) [inline .github/workflows/check.yaml:98]: same above

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: .github/workflows/check.yaml [hunk 3]
```diff
@@ -0,0 +1,121 @@
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind Srl
```
Comment (harlem88) [inline .github/workflows/check.yaml:3]:
```suggestion
# Copyright 2022-2024 SECO Mind Srl
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+# This file is part of Astarte.
+# Copyright 2023 SECO Mind Srl
```
Comment (harlem88) [inline .github/workflows/test.yaml:2]:
```suggestion
# This file is part of Astarte.
#
# Copyright 2023 SECO Mind Srl
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Review by rgallor — APPROVED** [review]

**Pull request merged**